kb/2023-08-21/upgrade-harvester-cloud-provider.md

```diff
@@ -0,0 +1,169 @@
+hide_table_of_contents: false
+---
+
+As the **Harvester v1.2.0** releases, a new Harvester cloud provider version **0.2.2** is integrated into the RKE2 **v1.27.3+rke2r1**, **v1.26.6+rke2r1**, **v1.25.11+rke2r1** and **v1.24.15+rke2r1**. 
+
+With Harvester v1.2.0, the new Harvester cloud provider provides more powerful load balancing for Kubernetes services. It modifies the network architecture, allowing the load balancer IP to be directly exposed in the RKE2 network. Refer to [the design document](https://github.com/harvester/harvester/blob/master/enhancements/20220214-harvester-cloud-provider-enhancement.md) for more details.
+
+If upgrading RKE2 to the version with the new Harvester cloud provider using **Rancher v2.7.6**, Rancher will pass the cluster name to the Harvester cloud provider(refer to [issue 4332](https://github.com/harvester/harvester/issues/4232)) . The Harvester cloud provider will use the cluster name as part of the Harvester load balancer name and will create a new load balancer for the original load balancer service because of the name change. This article will detail how to keep the original load balancer IP for the services by the following example.
+
+## Environment Prerequisites
+- RKE2 v1.24.10+rke2r1 `testing` in the VM network `default/mgmt-untagged` with two load balancer services 
+  - `default/lb0`: DHCP mode, with load balancer IP `172.19.105.215`
+  - `default/lb1`: Pool mode, with load balancer IP `192.168.100.2`
+
+- Harvester v1.2.0
+  - IP Pool `default` 
+    ``` yaml
+    apiVersion: loadbalancer.harvesterhci.io/v1beta1
+    kind: IPPool
+    metadata:
+      labels:
+        loadbalancer.harvesterhci.io/global-ip-pool: 'false'
+        loadbalancer.harvesterhci.io/vid: '0'
+      name: default
+    spec:
+      ranges:
+      - subnet: 192.168.100.0/24
+      selector:
+        scope:
+        - guestCluster: '*'
+          namespace: default
+          project: '*'
+    status:
+      allocated:
+        192.168.100.2: default/kubernetes-default-lb1-cd8c89e7
+      available: 252
+      lastAllocated: 192.168.100.2
+      total: 253
+    ```
+  - Tow Harvester load balancers
+    - `default/kubernetes-default-lb0-6df182ca` corresponds to the service `default/lb0` in the RKE2 cluster
+    - `default/kubernetes-default-lb1-cd8c89e7` corresponds to the service `default/lb1` in the RKE2 cluster
+
+- Rancher v2.7.6 to manage the Harvester and RKE2
+
+## Steps to Keep Load Balancer IP
+1. Scale the Harvester cloud provider deployment to 0 in the RKE2 cluster `testing`
+    ``` bash
+    kubectl -n kube-system scale deploy harvester-cloud-provider --replicas=0
+    ```
+2. For the service with DHCP mode, copy annotations `kube-vip.io/hwaddr` and `kube-vip.io/requestedIP` of the service which has the same name with the load balancer. In our example, copy the annotations from `default/kubernetes-default-lb0-6df182ca` in the Harvester cluster to the service `default/lb0` in the RKE2 cluster.
```

Comment:
And what's next, should users delete the existing svc and allow CCM to re-create a new one automatically, can u compete the description of this case?